src/lib/mina_generators/user_command_generators.ml

```diff
@@ -141,33 +143,32 @@ let sequence_parties_with_ledger ?max_other_parties ?length ?vk ?failure () =
     | None ->
         Quickcheck.Generator.small_non_negative_int
   in
+  let max_other_parties =
+    Option.value max_other_parties ~default:Parties_generators.max_other_parties
+  in
+  let num_keypairs = length * max_other_parties * 2 in
   (* Keep track of account states across multiple parties transaction *)
   let account_state_tbl = Account_id.Table.create () in
-  let merge_ledger source_ledger target_ledger =
-    (* add all accounts in source to target *)
-    Ledger.iteri source_ledger ~f:(fun _ndx acct ->
-        let acct_id = Account_id.create acct.public_key acct.token_id in
-        match Ledger.get_or_create_account target_ledger acct_id acct with
-        | Ok (`Added, _) ->
-            ()
-        | Ok (`Existed, _) ->
-            failwith "Account already existed in target ledger"
-        | Error err ->
-            failwithf "Could not add account to target ledger: %s"
-              (Error.to_string_hum err) () )
+  let%bind parties, fee_payer_keypair, keymap, ledger =
+    parties_with_ledger ~num_keypairs ~max_other_parties ~account_state_tbl ?vk
+      ?failure ()
   in
-  let init_ledger = Ledger.create ~depth:ledger_depth () in
   let rec go parties_and_fee_payer_keypairs n =
-    if n <= 0 then return (List.rev parties_and_fee_payer_keypairs, init_ledger)
+    if n <= 1 then return (List.rev parties_and_fee_payer_keypairs, ledger)
     else
-      let%bind parties, fee_payer_keypair, keymap, ledger =
-        parties_with_ledger ?max_other_parties ~account_state_tbl ?vk ?failure
-          ()
+      let%bind parties =
+        Parties_generators.gen_parties_from ~max_other_parties
+          ~fee_payer_keypair ~keymap ~ledger ~account_state_tbl ?vk ?failure ()
+      in
+      let valid_parties =
+        Option.value_exn
+          (Parties.Valid.to_valid ~ledger ~get:Ledger.get
+             ~location_of_account:Ledger.location_of_account parties )
       in
       let parties_and_fee_payer_keypairs' =
-        (parties, fee_payer_keypair, keymap) :: parties_and_fee_payer_keypairs
+        (User_command.Parties valid_parties, fee_payer_keypair, keymap)
+        :: parties_and_fee_payer_keypairs
       in
-      merge_ledger ledger init_ledger ;
       go parties_and_fee_payer_keypairs' (n - 1)
   in
-  go [] length
+  go [ (parties, fee_payer_keypair, keymap) ] length
```

Comment:
it's a little funny that you're not starting with the empty list.

if you need to distinguish this first tuple from the others, because it's generated in a different way, I think it would be more idiomatic to call `go` with the empty list, then cons the first tuple onto that result